controllers/cnbi/controller.go

```diff
@@ -42,7 +40,6 @@ import (
 type CustomNBImageReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
-	CNBi   *meteorv1alpha1.CustomNBImage
```

Comment:
why dont we want to have it as part of the reconciler type?

Reply:
Principle of least power. I re-used the example from https://sdk.operatorframework.io/docs/building-operators/golang/references/client/ (see "Reconcile and the Client API").

The rational is that client and runtime.Scheme are used in a different way than the CNBi object. They're not modified in sub-fonctions (`reconcilePipelineRun`), while it is.

Another option I didn't explore would be to make reconcilePipelineRun a receiving function on a CNBi value instead ; which would copy the object in order to return a modified version (if I understood go semantics correctly). But I think it's not a DeepCopy, so I'm not sure it would work correctly for various member of the type.